jwst/assign_wcs/tests/test_wcs.py

```diff
@@ -155,3 +158,108 @@ def test_create_fitswcs(tmpdir, create_model_3d):
+    'ctype2': 'DEC--TAN-SIP',
+    'crpix1': 1025.0,
+    'crpix2': 1025.0,
+    'crval1': 22.023508708126627,
+    'crval2': 11.998764139672561,
+    'cd1_1': -1.7436649443640038e-05,
+    'cd1_2': -2.0849272804435688e-08,
+    'cd2_1': -4.616114099568815e-08,
+    'cd2_2': 1.7519905482510155e-05,
+    'a_order': 3,
+    'b_order': 3,
+    'a_0_2': -1.5520006349224245e-06,
+    'a_0_3': 4.3714933936267264e-13,
+    'a_1_1': -1.1604291350572094e-05,
+    'a_1_2': 1.7125037791723067e-09,
+    'a_2_0': 1.9225112320436695e-06,
+    'a_2_1': -7.926931183344235e-11,
+    'a_3_0': 1.5974716218080076e-09,
+    'b_0_2': -6.7554035599749904e-06,
+    'b_0_3': 1.6911444547233376e-09,
+    'b_1_1': 3.6345928479129676e-06,
+    'b_1_2': -8.64927324063469e-11,
+    'b_2_0': 4.915411904251364e-06,
+    'b_2_1': 1.5701011999755348e-09,
+    'b_3_0': -2.9577497324389284e-12,
+    'sipmxerr': 3.950294659161568e-11,
+    'sipiverr': 0.11047793758675752,
+    'ap_0_1': 3.988114493629691e-07,
+    'ap_0_2': 1.5486628623564042e-06,
+    'ap_0_3': 3.784084622814559e-11,
+    'ap_1_0': -7.65706188115184e-06,
+    'ap_1_1': 1.149989280312782e-05,
+    'ap_1_2': -1.4994544929498265e-09,
+    'ap_2_0': -1.904283418222981e-06,
+    'ap_2_1': -4.507459210843236e-11,
+    'ap_3_0': -1.630572028632995e-09,
+    'bp_0_1': -7.46518175488653e-06,
+    'bp_0_2': 6.690996263911794e-06,
+    'bp_0_3': -1.5898038512674783e-09,
+    'bp_1_0': 4.524878821020556e-07,
+    'bp_1_1': -3.60126568682908e-06,
+    'bp_1_2': -4.4691036548847474e-11,
+    'bp_2_0': -4.9044713849991446e-06,
+    'bp_2_1': -1.7110976247273783e-09,
+    'bp_3_0': 3.895381500452847e-11,
+    'ap_order': 3,
+    'bp_order': 3}
+
+    data_path = os.path.split(os.path.abspath(data.__file__))[0]
+    test_file = os.path.join(data_path, 'jw42424001001_01101_00001_nrca5_rate.fits')
+
+    pipe = AssignWcsStep()
+    result = pipe.process(test_file)
+
+    # check that result.meta.wcsinfo has correct
+    # values after SIP approx.
+    wcs_info = result.meta.wcsinfo.instance
+
+    # make sure all expected keys are there
+    assert set(true_wcs.keys()).issubset(set((wcs_info.keys())))
+
+    # and make sure they match
+    for key in true_wcs:
+        try: # make sure numerical values match
+            k1 = float(true_wcs[key])
+            k2 = float(wcs_info[key])
+            assert_allclose(k1, k2)
+        except ValueError: # and str values match
+            assert true_wcs[key] == wcs_info[key]
+
+    # evaulate fits wcs and gwcs, make sure they agree
+    x = arange(10)
+    y = arange(10)
+    gwcs_ra, gwcs_dec = result.meta.wcs(x, y)
+    fits_wcs = wcs.WCS(wcs_info)
+    fitswcs_res = fits_wcs.pixel_to_world(x, y)
+
+    assert_allclose(fitswcs_res.ra.deg, gwcs_ra)
+    assert_allclose(fitswcs_res.dec.deg, gwcs_dec)
+            
+    #now write the file out, read it back in, and check that the fit values are preserved
+    result.write('tmp.fits')
+    result_read = ImageModel('tmp.fits')
```

Comment:
Use your pytest `tmpdir` fixture for this:

```python
path = str(tmpdir.join("tmp_sip_wcs.fits"))
result.write(path)
```
and use the `with` context manager to open:

```python
with datamodels.open(path) as result_read:
    assert result.meta.wcsinfo == result_read.meta.wcsinfo
```